lib/metrics/prometheus.js

```diff
@@ -6,27 +6,29 @@ const Prometheus = require('prom-client');
 class PrometheusMetric {
     constructor(options, client) {
         this.client = client;
-        if (options.labels === undefined) {
-            options.labels = { names: [] };
+        this.options = options;
+        if (this.options.labels === undefined) {
+            this.options.labels = { names: [] };
         }
-        this.staticLabels = options.prometheus.staticLabels;
+        this.staticLabels = this.options.prometheus.staticLabels;
         if (this.staticLabels !== undefined) {
             if (Object.keys(this.staticLabels).length > 0) {
                 Object.keys(this.staticLabels).forEach((name) => {
-                    options.labels.names.unshift(name);
+                    this.options.labels.names.unshift(name);
```

Comment:
This actually modifies the original labels and original options still. Everything is passed/assigned by reference.

Reply:
You can depend on https://www.npmjs.com/package/lodash.clonedeep and use it to create deep copies where needed. Like here it will be this.options = cloneDeep(options) and then you modify this.options like you did.

Reply:
Updated and tested with node 6